test/WebJobs.Script.Tests.Integration/Management/InstanceManagerTests.cs

```diff
@@ -501,28 +501,28 @@ public async Task Mounts_Valid_BYOS_Accounts()
 
             var meshInitServiceClient = new Mock<IMeshServiceClient>(MockBehavior.Strict);
             meshInitServiceClient.Setup(client =>
-                    client.MountCifs(Utility.BuildStorageConnectionString(account1, accessKey1), share1, targetPath1))
+                    client.MountCifs(Utility.BuildStorageConnectionString(account1, accessKey1, CloudConfig.CloudConstants.AzureStorageSuffix), share1, targetPath1))
                 .Throws(new Exception("Mount failure"));
             meshInitServiceClient.Setup(client =>
-                client.MountCifs(Utility.BuildStorageConnectionString(account2, accessKey2), share2, targetPath2)).Returns(Task.FromResult(true));
+                client.MountCifs(Utility.BuildStorageConnectionString(account2, accessKey2, CloudConfig.CloudConstants.AzureStorageSuffix), share2, targetPath2)).Returns(Task.FromResult(true));
             meshInitServiceClient.Setup(client =>
-                client.MountBlob(Utility.BuildStorageConnectionString(account3, accessKey3), share3, targetPath3)).Returns(Task.FromResult(true));
+                client.MountBlob(Utility.BuildStorageConnectionString(account3, accessKey3, CloudConfig.CloudConstants.AzureStorageSuffix), share3, targetPath3)).Returns(Task.FromResult(true));
 
             var instanceManager = new InstanceManager(_optionsFactory, _httpClient, _scriptWebEnvironment, _environment,
                 _loggerFactory.CreateLogger<InstanceManager>(), new TestMetricsLogger(), meshInitServiceClient.Object);
 
             instanceManager.StartAssignment(hostAssignmentContext, false);
 
-            await Task.Delay(TimeSpan.FromSeconds(0.5));
+            await Task.Delay(TimeSpan.FromSeconds(2));
```

Comment:
rather than a hardcoded delay, can you use the TestHelpers.Await to unblock early when the condition you're looking for is reached? Will also make the test less flaky if it ends up taking longer

Reply:
Didn't know we had something like this.